src/modules/silentpayments/main_impl.h

```diff
@@ -365,4 +365,349 @@ int secp256k1_silentpayments_recipient_create_labeled_spend_pubkey(const secp256
+    /* These functions should never fail at this point considering:
+     *   - loading the pubkey and input hash can only fail if the prevouts_summary object was created incorrectly
+     *     and we already check for this above.
+     *   - `_tweak_mul` can only fail if input_hash_scalar is zero, but assuming the prevouts_summary object
+     *     was created correctly, this is impossible because input_hash_scalar is the output of a hash function.
+     *   - `_eckey_pubkey_serialize` can only fail if the point we are trying to serialize is the point at infinity.
+     *
+     *   Note: we don't verify that the input hash is less than the curve order since this is verified when the
+     *   prevouts_summary object is created.
+     */
+    secp256k1_ge_from_bytes(&ge, &prevouts_summary->data[5]);
+    combined = (int)prevouts_summary->data[4];
+    ret = 1;
+    if (!combined) {
+        secp256k1_scalar input_hash_scalar;
+        secp256k1_scalar_set_b32(&input_hash_scalar, &prevouts_summary->data[5 + 64], NULL);
+        ret &= secp256k1_eckey_pubkey_tweak_mul(&ge, &input_hash_scalar);
+    }
+    ret &= secp256k1_eckey_pubkey_serialize(&ge, output33, &pubkeylen, 1);
+#ifdef VERIFY
+    VERIFY_CHECK(ret && pubkeylen == 33);
+#else
+    (void)ret;
+#endif
+    return 1;
+}
+
+int secp256k1_silentpayments_recipient_prevouts_summary_parse(const secp256k1_context *ctx, secp256k1_silentpayments_prevouts_summary *prevouts_summary, const unsigned char *input33) {
+    size_t inputlen = 33;
+    secp256k1_ge pk;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(prevouts_summary != NULL);
+    ARG_CHECK(input33 != NULL);
+    /* Since an attacker can send us malicious data that looks like a serialized public key but is not, fail early. */
+    if (!secp256k1_eckey_pubkey_parse(&pk, input33, inputlen)) {
+        return 0;
+    }
+    /* A serialized prevouts_summary will always have have the input_hash multiplied in, so we set combined = true.
+     * Additionally, we zero out the 32 bytes used to represent the input_hash.
+     */
+    memcpy(&prevouts_summary->data[0], secp256k1_silentpayments_prevouts_summary_magic, 4);
+    prevouts_summary->data[4] = 1;
+    secp256k1_ge_to_bytes(&prevouts_summary->data[5], &pk);
+    memset(&prevouts_summary->data[5 + 64], 0, 32);
+    return 1;
+}
+
+int secp256k1_silentpayments_recipient_scan_outputs(
+    const secp256k1_context *ctx,
+    secp256k1_silentpayments_found_output **found_outputs, size_t *n_found_outputs,
+    const secp256k1_xonly_pubkey * const *tx_outputs, size_t n_tx_outputs,
+    const unsigned char *scan_key32,
+    const secp256k1_silentpayments_prevouts_summary *prevouts_summary,
+    const secp256k1_pubkey *spend_pubkey,
+    const secp256k1_silentpayments_label_lookup label_lookup,
+    const void *label_context
+) {
+    secp256k1_scalar output_tweak_scalar, rsk_scalar;
```

Comment:
nit, here and in the `_create_shared_secret` function: could rename `rsk_scalar` to something more explicit like `scan_key_scalar` (even after many review rounds, I tend to repeatedly forget that `rsk` is short for "recipient scan key")